python/cornserve/cli/tasklib_explorer.py

```diff
@@ -0,0 +1,127 @@
+        ):
+            out.append("-")
+        out.append(ch.lower())
+    return "".join(out)
+
+
+def discover_tasklib() -> tuple[
+    list[dict[str, Any]],
+    list[dict[str, Any]],
+    list[dict[str, Any]],
+]:
+    """Discover unit/composite tasks and descriptors from cornserve_tasklib.
+
+    Returns:
+        A tuple of (unit_task_entries, composite_task_entries, descriptor_entries),
+        where each item is a list of dictionaries ready to be serialized into the
+        gateway deployment request payloads.
+    """
+    try:
+        import cornserve_tasklib  # noqa: F401  # ensure package importable
+        from cornserve.task.base import Task, UnitTask
+        from cornserve.task_executors.descriptor.base import TaskExecutionDescriptor
+    except Exception as e:  # pragma: no cover - bubbled to CLI
+        raise ImportError(f"Failed to import cornserve_tasklib or cornserve core modules: {e}") from e
+
+    unit_task_entries: list[dict[str, Any]] = []
+    composite_task_entries: list[dict[str, Any]] = []
+    descriptor_entries: list[dict[str, Any]] = []
+
+    module_source_cache: dict[str, str] = {}
+
+    def get_module_source_b64(module) -> str:
+        if module.__name__ not in module_source_cache:
+            src = inspect.getsource(module)
+            module_source_cache[module.__name__] = base64.b64encode(src.encode("utf-8")).decode("ascii")
+        return module_source_cache[module.__name__]
+
+    # Discover tasks (unit and composite)
+    task_pkg = importlib.import_module("cornserve_tasklib.task")
+    for modinfo in pkgutil.walk_packages(task_pkg.__path__, prefix=task_pkg.__name__ + "."):
+        module = importlib.import_module(modinfo.name)
+        for name, obj in inspect.getmembers(module, inspect.isclass):
+            if obj.__module__ != module.__name__:
+                continue
+            if issubclass(obj, UnitTask) and obj is not UnitTask:
+                unit_task_entries.append(
+                    {
+                        "source_b64": get_module_source_b64(module),
+                        "task_class_name": obj.__name__,
+                        "task_definition_name": _camel_to_kebab(obj.__name__),
+                        "module_name": module.__name__,
+                        "is_unit_task": True,
+                    }
+                )
+            elif issubclass(obj, Task) and (not issubclass(obj, UnitTask)) and obj is not Task:
+                composite_task_entries.append(
+                    {
+                        "source_b64": get_module_source_b64(module),
+                        "task_class_name": obj.__name__,
+                        "task_definition_name": _camel_to_kebab(obj.__name__),
+                        "module_name": module.__name__,
+                        "is_unit_task": False,
+                    }
+                )
+
+    # Discover descriptors
+    desc_pkg = importlib.import_module("cornserve_tasklib.task_executors.descriptor")
```

Comment:
Why two levels? Is there something inside `tasklib.task_executors` other than `descriptor`?